js/5etools-vehicles.js

```diff
@@ -0,0 +1,230 @@
+						character.attribs.create({name: "vehicle_dt", current: achpContainer.dt || ""});
+						character.attribs.create({name: "vehicle_keel", current: keel});
+						character.attribs.create({name: "vehicle_beam", current: beam});
+						character.attribs.create({name: "vehicle_speed", current: speed});
+						character.attribs.create({name: "vehicle_cost", current: Parser.vehicleCostToFull(data)});
+
+
+						// This adds weapons, or anything that might end up in the weapon catagory
+						if (data.weapon) d20plus.importer.addVehicleWeapons(character, data.weapon, renderer);
+						if (data.control) d20plus.importer.addVehicleWeapons(character, data.control, renderer, "Control");
+						if (data.movement) d20plus.importer.addVehicleWeapons(character, data.movement, renderer, "Movement");
+						if (data.trait) d20plus.importer.addVehicleWeapons(character, data.trait, renderer, "Trait");
+
+						let numActions = 0;
+
+						// Some ships (mostly UA) have an other catagory that contains an action explaining how that ship's actions work
+						if (data.other) {
+							d20plus.vehicles._addAction(character, renderer, data.other[0].name, data.other[0].entries, numActions++);
+						}
+						// Mostly GoS ships
+						if (data.action) {
+							const text = d20plus.importer.getCleanText(renderer.render(data.action[0], 1)).replace(/^\s*Hit:\s*/, "");
+							d20plus.importer.addVehicleAction(character, "Actions", text, numActions++);
+
+							data.action[1].items.forEach((a, i) => {
+								let actionEntries = [];
+								if (data.weapon && i < data.weapon.length) actionEntries = data.weapon[i].entries;
+								actionEntries.push(a.entry);
+								d20plus.vehicles._addAction(character, renderer, a.name, actionEntries, numActions++);
+							});
+						}
+						// Mostly DiA war machines
+						else if (data.actionStation) {
+							data.actionStation.forEach(a => {
+								d20plus.vehicles._addAction(character, renderer, a.name, a.entries, numActions++);
+							});
+						}
+						// Sort of a catch all but works especially well with spelljammer
+						else if (data.weapon) {
+							data.weapon.forEach(w => {
+								if (w.action) {
+									w.action.forEach(a => {
+										d20plus.vehicles._addAction(character, renderer, a.name, a.entries.concat(w.entries), numActions++);
+									});
+								}
+								else if (w.entries) {
+									d20plus.vehicles._addAction(character, renderer, w.name, w.entries, numActions++);
+								}
+							});
+						}
+
+						character.view.updateSheetValues();
+
+						// Deal with fluff and images here
+						const fluff = d20plus.vehicles.fluff?.find(it => it.name === data.name && it.source === data.source);
+						d20plus.vehicles._setFluffImage(character, data, fluff);
+						d20plus.vehicles._setFluff(character, data, fluff, renderer);
+
+
+					} catch (e) {
+						d20plus.ut.log(`Error loading [${name}]`);
+						d20plus.monsters.addImportError(name);
```

Comment:
is `d20plus.monsters` a copy-paste typo? (Or, should the method be moved somewhere more general? 🙃)